### ISSUE TITLE
[Markdown] Integrate Image-Support

### ISSUE THREAD

**Issue by Bizarrus** [dlang/dub-registry#578]
Images will be not shown.

**Original:**
![grafik](https://github.com/user-attachments/assets/5f7b6f81-b1f9-459d-9834-4dbf39dd2953)

**Package-Site:**
![grafik](https://github.com/user-attachments/assets/d96b2b36-5187-4c51-8053-c964965ea64c)

**Example:**

https://code.dlang.org/packages/coldbite?tab=info

**Comment by s-ludwig** [comment]
I'm not sure why GitHub accepts the syntax, but instead of `[![DOCUMENTATION]](https://github.com/coldbite/Engine/wiki)`, `![DOCUMENTATION](https://github.com/coldbite/Engine/wiki)` should work and is what is expected by the parser. See the [CommonMark spec](https://spec.commonmark.org/0.31.2/#images) for reference.

**Comment by Bizarrus** [comment]
Because, it is **nested**.

```![alt](src)``` = ```<img alt="$alt" src="$src">```

```[text](href)``` = ```<a href="$href">$href</a>```

**text** will be replaced with the image:
```[![src]](href)``` = ```<a href="$href"><img src="$src"></a>```

**Comment by Bizarrus** [comment]
> See the [CommonMark spec](https://spec.commonmark.org/0.31.2/#images) for reference.
Please study the reference:

> [!NOTE]
> The link text may contain inline content:
> ![IMG-20241124-WA0001](https://github.com/user-attachments/assets/bb35f8a9-a8f3-4131-b0bf-c30870185b46)

**Comment by s-ludwig** [comment]
> Please study the reference, when you link it:

Sorry, I fail to see how the image you provided applies to the code you've written. Can you enlighten me?

**Comment by Bizarrus** [comment]
You can reference Images an Links like:
```
[refImage]: http://example.cpm/image.png
```

Then you can use the references:
```
[![refImage]](http://example.com)
```

Thats the same like:
```
[![alt](http://example.com/image.png)](http://example.com)
```

**Comment by s-ludwig** [comment]
A reference works like this: `![alt text][refImage]`

**Comment by s-ludwig** [comment]
Also note that an `alt` text in mandatory as per the HTML spec..

**Comment by Bizarrus** [comment]
> A reference works like this: `![alt text][refImage]`

No, you can't use the Syntax `[a][b]` or `![a][b]`. Some other attributes will be set in the reference.

See the example for Links:
![grafik](https://github.com/user-attachments/assets/a3b555b7-e05b-43e2-bbaa-bebb738ad090)


Here is a live example of Images:

**Example:**
[![imageRef]](http://example.com/)

[imageRef]: https://placehold.co/100x100 "alt"

**Code:**
```
[![imageRef]](http://example.com/)

[imageRef]: https://placehold.co/100x100 "title"
```

**Rendered:**
```
<a href="http://example.com/" rel="nofollow">
    <img src="https://placehold.co/100x100" alt="imageRef" title="title" data-canonical-src="https://placehold.co/100x100" style="max-width: 100%;">
</a>
```

The **alt** attribute is automatically on image references the reference name.

**Comment by s-ludwig** [comment]
This has nothing to do with the references at the bottom and it has nothing to do with the nesting. The problem is the shortcut syntax you are using (`![refImage]` instead of `![alt text][refImage]`). This is simply not supported by the parser, whereas images, nested links/images and link references are all handled fine.

Supporting this particular syntax, as well as many other details of the CommonMark spec, is certainly a noble goal, but please understand that your issue is addressing the wrong subject.
